src/agent/platform.rs

```diff
@@ -136,172 +141,133 @@ impl Client {
 
 #[derive(Clone)]
 struct AuthToken {
-    token: Arc<Mutex<AsciiMetadataValue>>,
+    inner: Arc<Mutex<AsciiMetadataValue>>,
 }
 
 impl AuthToken {
-    fn new(val: &str) -> Result<Self> {
-        let token = format_bearer(val)?;
+    fn new(token: &str) -> Self {
+        let bearer = format_bearer(token);
 
-        Ok(Self {
-            token: Arc::new(Mutex::new(token)),
-        })
+        Self {
+            inner: Arc::new(Mutex::new(bearer)),
+        }
     }
 
-    fn get(&self) -> AsciiMetadataValue {
-        self.token.lock()
-            .unwrap()
-            .clone()
+    fn bearer(&self) -> AsciiMetadataValue {
+        self.inner.lock().unwrap().clone()
     }
 
-    fn set(&self, val: &str) -> Result<()> {
-        let mut lk = self.token.lock().unwrap();
-        *lk = format_bearer(val)?;
-        Ok(())
+    fn set(&self, token: &str) {
+        *self.inner.lock().unwrap() = format_bearer(token);
     }
 }
 
-fn format_bearer(val: &str) -> Result<AsciiMetadataValue> {
-    Ok(format!("Bearer {val}").parse()?)
+impl Interceptor for AuthToken {
+    fn call(&mut self, mut request: Request<()>) -> std::result::Result<Request<()>, Status> {
+        request
+            .metadata_mut()
+            .insert("authorization", self.bearer());
```

Comment:
This doesn't need to be capitalized?

Reply:
Actually the new thing is lower case http headers. I think it's more or less required for http2, which is what grpc uses.